libpod/sqlite_state.go

```diff
@@ -71,7 +67,17 @@ func NewSqliteState(runtime *Runtime) (_ State, defErr error) {
 		return nil, fmt.Errorf("creating root directory: %w", err)
 	}
 
-	conn, err := sql.Open("sqlite3", filepath.Join(basePath, sqliteOptions))
+	// Make sure busy timeout is set to high value to keep retrying when the db is locked.
+	// Timeout is in ms, so set it to 100s to have enough time to retry the operations.
+	// Some users might want to experiment with different timeout values (#23236)
+	// DO NOT DOCUMENT or recommend PODMAN_SQLITE_BUSY_TIMEOUT outside of testing.
+	busyTimeout := "100000"
+	if env, ok := os.LookupEnv("PODMAN_SQLITE_BUSY_TIMEOUT"); ok {
+		busyTimeout = env
+	}
+	sqliteOptionBusyTimeout := "&_busy_timeout=" + busyTimeout
+
+	conn, err := sql.Open("sqlite3", filepath.Join(basePath, sqliteOptions+sqliteOptionBusyTimeout))
```

Comment:
Do we get a vaguely sane error if the user enters something that is not a number? I have visions of someone having this environment variable set to a garbage value and having a hell of a time debugging that.

Reply:
```
Error: beginning transaction: Invalid _busy_timeout: notInt: strconv.ParseInt: parsing "notInt": invalid syntax
```

Reply:
Perfect, thank god for SQLite's excellent error messages